From 41ff7a2796a019c566d0a27898791ca85348b843 Mon Sep 17 00:00:00 2001
From: Carlos O'Ryan <coryan@google.com>
Date: Sun, 6 Feb 2022 13:34:45 +0000
Subject: [PATCH 2/2] Add explicit conversion

Explicitly convert from `std::unique_ptr<Buffer>` to a `std::shared_ptr` on the way to convert to `Result<std::shared_ptr<>>`.  I am not sure why it only failed with one compiler.
---
 cpp/src/arrow/filesystem/gcsfs.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cpp/src/arrow/filesystem/gcsfs.cc b/cpp/src/arrow/filesystem/gcsfs.cc
index a61fef224..1f719c3ff 100644
--- a/cpp/src/arrow/filesystem/gcsfs.cc
+++ b/cpp/src/arrow/filesystem/gcsfs.cc
@@ -138,7 +138,7 @@ class GcsInputStream : public arrow::io::InputStream {
     stream_.read(reinterpret_cast<char*>(buffer->mutable_data()), nbytes);
     ARROW_GCS_RETURN_NOT_OK(stream_.status());
     RETURN_NOT_OK(buffer->Resize(stream_.gcount(), true));
-    return buffer;
+    return std::shared_ptr<Buffer>(std::move(buffer));
   }
   //@}
 
-- 
2.37.0.windows.1

